### ISSUE TITLE
feature request:  FLOSS build

### ISSUE THREAD

**Issue by magnetophon** [ech2/airwindows-ports#1]
Hi,

Many thanks for making this!
I would love to use these wonderfull plugins, but I really want to keep using FLOSS sw only.

Any chance of a build that doesn't need the steinberg vst-sdk?
Possible options are:
- a build based on [vestige](https://github.com/LMMS/lmms/tree/master/plugins/vestige)
- an lv2 port
- ???

Many thanks again!

**Comment by ech2** [comment]
Hey,

I can’t promise anything right now, as I have little spare time lately. Could you please explain, what vestige does? Is it a wrapper around wine or something?

Cheers

**Comment by magnetophon** [comment]
Thanks for the quick response!
Vestige is a way to build vst's for linux, without needing the steinberg headers.

**Comment by x42** [comment]
vestige is a clean-room reverse-engineered  VST-SDK which is not license encumbered.

As you may know the official Steinberg VST2 SDK is no longer available, cannot be redistributed andis also not compatible with free-software licenses.

**Comment by ech2** [comment]
OK, thanks for the explanations. So, Vestige is a drop-in replacement for VST SDK, right? If this is so, it’d be easy to do.

Regarding licenses, I thought Steinberg added GPLv3 as an option to their SDK. Or is it only for VST3? I think I used vst36-sdk package from AUR to build this package, but don’t really know much about licensing.

Disclaimer: I don’t have much experience in audio programming and open source.

**Comment by x42** [comment]
vestige is not a drop-in replacement. it was clean-room reverse engineered. it describes the same interface but it as a different API.

Anyway, since the airwindows plugins are just DSP + ctrl ports.  wrapping them into any plugin-wrapper is mostly trivial.

**Comment by x42** [comment]
PS. https://drobilla.net/software/mda-lv2  has a wrapper for VST AudioEffectX to LV2 which is pretty much drop-in.

**Comment by ech2** [comment]
Thanks @x42! I’ll have a look after I sort out my things a little. Btw, can I use a vestige-compiled plugin in, say, Reaper, which is closed source and only has VST.

**Comment by x42** [comment]
>  Btw, can I use a vestige-compiled plugin in, say, Reaper, which is closed source and only has VST.

Yes, you can. VST2 is just an interface/API (basically a header without implementation). Once you compiled a plugin you cannot tell the difference.  JUCE is another example that can produce VST2 without the official Steinberg SDK.

As far as I can tell the official Steinberg SDK has some convenience C++ classes that makes implementing plugins a lot simpler and those C++ classes do include copyrighted implementation.

**Comment by magnetophon** [comment]
Another option would be [the GPL'd vst3sdk](https://github.com/steinbergmedia/vst3sdk).

**Comment by x42** [comment]
> Another option would be the GPL'd vst3sdk.

Except there's currently no free/libre host on GNU/Linux that features VST3 support.

**Comment by magnetophon** [comment]
Oh... :(

Thanks for the quick info!

**Comment by magnetophon** [comment]
Any news on this?
